src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,35 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        switch (type.getSimpleName()) {
+            case "Truck":
+                return new TruckProducer().get();
+            case "Bulldozer":
+                return new BulldozerProducer().get();
+            case "Excavator":
+                return new ExcavatorProducer().get();
+            default:
+                return new ArrayList<>();
```

Comment:
This method must be a good example of polymorphism. Let's call method get only once

Reply:
I hope, i understood you)